core/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy

```diff
@@ -67,14 +67,20 @@ grant codeBase "${codebase.mocksocket-1.2.jar}" {
 grant codeBase "${codebase.elasticsearch-rest-client-6.0.0-beta1-SNAPSHOT.jar}" {
   // rest makes socket connections for rest tests
   permission java.net.SocketPermission "*", "connect";
+  // rest client uses system properties which gets the default proxy
+  permission java.net.NetPermission "getProxySelector";
 };
 
 grant codeBase "${codebase.httpcore-nio-4.4.5.jar}" {
   // httpcore makes socket connections for rest tests
   permission java.net.SocketPermission "*", "connect";
+  // rest client uses system properties which gets the default proxy
+  permission java.net.NetPermission "getProxySelector";
```

Comment:
the call happens from `httpasyncclient` so we should be ok with omitting this permission grant to `httpcore-nio`